hedera-node/src/main/java/com/hedera/services/utils/accessors/SignedTxnAccessor.java

```diff
@@ -128,13 +134,27 @@ public static SignedTxnAccessor uncheckedFrom(Transaction validSignedTxn) {
 
     public static SignedTxnAccessor from(byte[] signedTxnWrapperBytes)
             throws InvalidProtocolBufferException {
-        return new SignedTxnAccessor(signedTxnWrapperBytes);
+        return new SignedTxnAccessor(signedTxnWrapperBytes, Optional.empty());
     }
 
-    protected SignedTxnAccessor(byte[] signedTxnWrapperBytes)
+    public static SignedTxnAccessor from(
+            byte[] signedTxnWrapperBytes, final Transaction signedTxnWrapper)
+            throws InvalidProtocolBufferException {
+        return new SignedTxnAccessor(signedTxnWrapperBytes, Optional.of(signedTxnWrapper));
+    }
+
+    protected SignedTxnAccessor(
+            byte[] signedTxnWrapperBytes, final Optional<Transaction> transaction)
```

Comment:
Slightly prefer using a `@Nullable` second argument here, since we don't gain much expressiveness by introducing the `Optional`.

Reply:
Fixed!